flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointListener.java

```diff
@@ -38,4 +38,12 @@
 	 * @throws Exception
 	 */
 	void notifyCheckpointComplete(long checkpointId) throws Exception;
+
+	/**
+	 * This method is called as a notification once a distributed checkpoint has been aborted.
+	 *
+	 * @param checkpointId The ID of the checkpoint that has been aborted.
+	 * @throws Exception
+	 */
+	default void notifyCheckpointAborted(long checkpointId) throws Exception {}
```

Comment:
I think this could still stay at `CheckpointListener.java` as this method could take effect in many places. Some state backend could leverage this method to do some cleanup things, especially those disAgg (separating storage and computation) state-backends. Moreover, we also want to introduce `notifyCheckpointSubsume` if possible in future, and too many interfaces would be a bit over design.

Reply:
> Some state backend could leverage this method to do some cleanup things, especially those disAgg (separating storage and computation) state-backends. 

If any of state backends will implement it, it won't be too difficult to add a call.

> too many interfaces would be a bit over design.

I think it's better to have many small interfaces rather than a bigger one (I believe it's a general recommendation).
It's also not over design here because this is an important part of the system.

As it is now, there is a lot of "dead" code with empty methods which complicates the design.

Reply:
As there are already such state backends, as @Myasuka mentioned below, it makes sense
(regarding multiple interfaces, they can simply be combined (with `extends`) and given to the user).